### ISSUE TITLE
smartypants

### ISSUE THREAD

**Issue by stsouko** [singingwolfboy/flask-misaka#18]
smartypants do PRE-process, but in documentation:
```Post-process rendered markdown text with SmartyPants.```

this break tables and over extentions.

**Comment by singingwolfboy** [comment]
@stsouko Could you please provide an example? Please include:
* How you configure Flask-Misaka
* The Markdown you pass it
* The HTML that you _expect_ to see as a result
* The HTML that you _actually_ see as a result

Can you also take a look at #19, and see if it resolves the issue the way you expect?

**Comment by stsouko** [comment]
Misaka(app, underline=True, math=True, strikethrough=True, superscript=True, tables=True, footnotes=True, smartypants=True)

smartypants break tables markup.
|---|---:|---:|

because it replace  ---  with &mdash

correctly smartypants have to do post-process.

**Comment by mindv0rtex** [comment]
I don't know how to make a correct pull-request, but this is fixed by a slight reshuffling of the `markdown` function:
```python
def markdown(text, renderer=None, **options):
    """
    Parses the provided Markdown-formatted text into valid HTML, and returns
    it as a :class:`flask.Markup` instance.

    :param text: Markdown-formatted text to be rendered into HTML
    :param renderer: A custom misaka renderer to be used instead of the default one
    :param options: Additional options for customizing the default renderer
    :return: A :class:`flask.Markup` instance representing the rendered text
    """
    ext, rndr = make_flags(**options)
    if renderer:
        md = misaka.Markdown(renderer, ext)
        result = md(text)
    else:
        result = misaka.html(text, extensions=ext, render_flags=rndr)
    if options.get("smartypants"):
        result = misaka.smartypants(result)
    return Markup(result)
```

This ensures that smartypants is applied after the md->html conversion

**Comment by singingwolfboy** [comment]
This is now fixed, and published in Flask-Misaka 1.0.0.
